### DISCUSSION TITLE
Why does strings.any_prefix_match failed to match an input

### DISCUSSION THREAD

**Discussion by ylbeethoven** [open-policy-agent/community#581] [OPA and Rego] [answered]
Hi,

I am trying to understand why below rego can't pass the validation on rego playground
```rego
package envoy.authz

import rego.v1

allow_path_prefix := "/a"

allow if {
	strings.any_prefix_match(input.attributes.request.http.path, allow_path_prefix)
}
```

 when the input is
```rego
{
    "input": {
        "attributes": {
            "request": {
                "http": {
                    "path": "/anything"
                }
            }
        }
    }
}
```
It works if I use the path string

```rego
package envoy.authz

import rego.v1

allow_path_prefix := "/a"

allow if {
	strings.any_prefix_match("/anything", allow_path_prefix)
}
```

isn't `input.attributes.request.http.path` the same value as `/anything`?

**Comment by johanfylling** [comment, marked as answer]
The Rego playground doesn't expect an outer `input` attribute in the provided input JSON document; this is something you use when e.g. passing parameters to the [REST query API](https://www.openpolicyagent.org/docs/latest/rest-api/#query-api), but not to the playground.

If you change the input JSON to 

```json
{
    "attributes": {
        "request": {
            "http": {
                "path": "/anything"
            }
        }
    }
}
```

as in: https://play.openpolicyagent.org/p/9CN3VEixkU, your code should evaluate as expected.

**Reply by ylbeethoven** [reply]
ah, i made this mistake again... thanks... 🥲
